portfolio/src/main/webapp/script.js

```diff
@@ -50,20 +50,105 @@ async function deleteData() {
 
 // Functions for Google Maps
 // Make the map
+let map;
+let editMarker;
+
 function createMap() {
-  console.log("createMap() exexuted");
+  console.log('createMap() exexuted');
   const locGoogleplex = {lat: 37.422, lng: -122.084};
+  const labels = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
   const mapInit = {
     center: locGoogleplex,
     zoom: 16}
-
-  const map = new google.maps.Map(
-    document.getElementById("map"),
+  // Initialize the map with <mapInit> values
+  map = new google.maps.Map(
+    document.getElementById('map'),
     mapInit);
-  console.log("createMap() finished");
 
+  // Hardcode a marker onto the map
   const markerGoogleplex = new google.maps.Marker({
     position: locGoogleplex,
     map: map,
     title: 'Googleplex office'});
+
+  // User clicks on map to pop up marker-creation-text-prompt
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+  // Get saved markers from datastore
+  fetchMarkers();
+}
+// Helper: Fetch saved markers from database and adds them to map
```

Comment:
nit: Typically you separate your end of the function and start of another by a newline. Here and below.